src/backend/tests/unmocked/recruitment.test.ts

```diff
@@ -88,4 +89,41 @@ describe('Recruitment Tests', () => {
       expect(result).toStrictEqual([milestone1, milestone2]);
     });
   });
+
+  describe('Delete a single milestone', () => {
+    it('Fails if user is not admin', async () => {
+      await expect(
+        async () => await RecruitmentServices.deleteMilestone(await createTestUser(wonderwomanGuest, orgId), 'id', orgId)
+      ).rejects.toThrow(new AccessDeniedAdminOnlyException('delete milestone'));
+    });
+
+    it('Fails if milestoneId is not found', async () => {
+      await expect(
+        async () => await RecruitmentServices.deleteMilestone(await createTestUser(batmanAppAdmin, orgId), 'id1', orgId)
+      ).rejects.toThrow(new HttpException(400, 'Milestone with id: id1 not found!'));
+    });
+
+    it('Fails if milestone is already deleted', async () => {
+      const testSuperman = await createTestUser(supermanAdmin, orgId);
+      const testMilestone = await createTestMileStone(testSuperman, orgId);
+      await RecruitmentServices.deleteMilestone(testSuperman, testMilestone.milestoneId, orgId);
+
+      await expect(
+        async () => await RecruitmentServices.deleteMilestone(testSuperman, testMilestone.milestoneId, orgId)
+      ).rejects.toThrow(new DeletedException('Milestone', testMilestone.milestoneId));
+    });
+
+    it('Succeeds and deletes milestone', async () => {
+      const testSuperman = await createTestUser(supermanAdmin, orgId);
+      const [testMilestone1] = await Promise.all([createTestMileStone(testSuperman, orgId)]);
```

Comment:
there is no need to make this an array and use promise.all. instead do it how you did it above on line 108